src/models/auth.js

```diff
@@ -12,9 +14,25 @@ export default {
         payload,
       });
     },
-    *logoutUser({ payload }, { put }) {
+    *loginUser({ payload }, { call }) {
+      const response = yield call(queryLoginUser, payload);
+      localStorage.setItem('token', response.auth_token);
+      return response;
+    },
+    *registerUser({ payload }, { call }) {
+      const response = yield call(queryRegisterUser, payload);
+      return response;
+    },
+    *logoutUser({ payload }, { call }) {
+      const response = yield call(queryLogoutUser, payload);
+      // TODO: Must be removed from the backend through
+      // the response header.
```

Comment:
I'm not sure what "response header" has to do with this; but, yes, we need to make the `logout` call.

What we do on failure is an interesting question; while it would probably be good for the user to know that the authorization token wasn't invalidated, with some sort of error (warning?) popup, we might want to go ahead and remove the browser token to terminate the session?

Reply:
Logout failure is definitely weird.  If, afterwards, the client and server still have valid, functioning tokens, I would not be in a hurry to dump the client's copy.  From a user perspective, (I think) I have pretty good control over my end of the dialog (e.g., I can close the browser, or reboot the system, if needed...); what I'm concerned about is ensuring that my context on the server does not become available to anyone else.  So, if I cannot log out of the server, I might want to retain my token so that I can try to take some sort of alternative action.

Perhaps we can move forward by dividing the problem.  We need a clear response from the server which indicates whether the token is still valid.  If it's not valid, then, in effect (from the client's perspective) the logout has succeeded, even though the result of the operation was reported as a failure, and we can remove the local copy of the token; however, if the token is still valid, then...I'm not sure what we do, but I don't see how dumping it from the client helps.